pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/containerd/logrotate/logrotate.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logrotate
+
+import (
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"k8s.io/utils/pointer"
+)
+
+// Config returns the content for logrotate units and files.
+// Whenever logrotate is ran, this config will:
+//   - keep only 14 old (rotated) logs, and will discard older logs.
+//
+// prefix carries the target container runtime such as (containerd, docker)
+// In containerd case the log rotation based on size is implemented by kubelet via container log manager
+// PR: https://github.com/kubernetes/kubernetes/pull/59898
+func Config(pathConfig, pathLogFiles, prefix string) ([]extensionsv1alpha1.Unit, []extensionsv1alpha1.File) {
```

Comment:
After you split the logrotate handling to 2 different packages, do you still need the `prefix` var? I assume previously it was passed from outside because there was only 1 logrotate handling. Now there are 2 pkgs and maybe you can drop this var and hard-code the usages of prefix with the container runtime type.